tests/acceptance/settings-page/billing-test.js

```diff
@@ -0,0 +1,149 @@
+import { module, test } from 'qunit';
+import { setupApplicationTest } from 'codecrafters-frontend/tests/helpers';
+import { signIn, signInAsSubscriber, signInAsVipUser } from 'codecrafters-frontend/tests/support/authentication-helpers';
+import testScenario from 'codecrafters-frontend/mirage/scenarios/test';
+import billingPage from 'codecrafters-frontend/tests/pages/settings/billing-page';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { setupWindowMock } from 'ember-window-mock/test-support';
+import percySnapshot from '@percy/ember';
+
+module('Acceptance | settings-page | billing-test', function (hooks) {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+  setupWindowMock(hooks);
+
+  test('membership section shows correct plan for subscriber with active subscription', async function (assert) {
+    testScenario(this.server);
+    signInAsSubscriber(this.owner, this.server);
+    const subscription = this.server.schema.subscriptions.first();
+    subscription.update('pricingPlanName', 'Yearly Plan');
+
+    await billingPage.visit();
+
+    assert.ok(billingPage.membershipSection.isVisible, 'membership section is visible');
+    assert.ok(billingPage.membershipSection.text.includes('Membership active'), 'shows active plan');
+
+    await percySnapshot('Billing Page - Active Subscription');
+  });
+
+  test('membership section shows correct plan for non-subscriber', async function (assert) {
+    testScenario(this.server);
+    signIn(this.owner, this.server);
+
+    await billingPage.visit();
+
+    assert.ok(billingPage.membershipSection.isVisible, 'membership section is visible');
+    assert.notOk(billingPage.membershipSection.hasActivePlan, 'does not show active plan');
+    await percySnapshot('Billing Page - No Active Subscription');
+  });
+
+  test('membership section shows VIP access for subscriber with VIP access', async function (assert) {
+    testScenario(this.server);
+    signInAsVipUser(this.owner, this.server);
+
+    await billingPage.visit();
+
+    assert.ok(billingPage.membershipSection.isVisible, 'membership section is visible');
+    assert.ok(billingPage.membershipSection.text.includes('VIP Access'), 'shows VIP access');
+    await percySnapshot('Billing Page - VIP Access');
+  });
+
+  test('support section is visible', async function (assert) {
+    testScenario(this.server);
+    signInAsSubscriber(this.owner, this.server);
+
+    await billingPage.visit();
+
+    assert.ok(billingPage.supportSection.isVisible, 'support section is visible');
+    assert.ok(billingPage.supportSection.clickOnContactButton, 'click on contact button');
+    assert.dom('[data-test-support-section]').hasAttribute('href', /^mailto:hello@codecrafters.io/);
```

Comment:
Isn't this just an attribute on the button? Why are we asserting against `data-test-support-section` here?

Reply:
Also, page objects do support checking attributes: https://ember-cli-page-object.js.org/docs/v2.3.x/api/attribute. So you could use that to define something like `contactButtonLink` or `contactButtonHref`